design/external-compaction.md

```diff
@@ -0,0 +1,249 @@
+
+  * *CompactionPlanner*: an object that is configured to create CompactionExecutors and determine which one to use for compacting a Tablet.
+
+  * *CompactionExecutor*: an object that contains some number of configurable Threads in which to run compactions
+
+  * *Compaction Dispatcher*: a per table pluggable component that decides which compaction service tablets will use for different kinds of compactions.  For example it may dispatch USER compactions to one service and SYSTEM compactions to another. 
+
+## Overview
+
+The CompactionManager operates on the set of online hosted tablets and calls CompactionManager::compact which uses the Table's configurable Compaction Dispatcher to determine which CompactionService to use for compacting the tablet for the specified type of operation (USER, SYSTEM, etc). Once the CompactionService is determined, the CompactionManager calls CompactionService::compact passing in the Tablet information and type of compaction. The CompactionService uses the CompactionPlanner to determine the plan (which files need to be compacted, which queue to use based on file sizes, etc), then executes the plan using CompactionExecutors. 
+
+There are short-circuits along this path when compactions are already queued for the tablet and other cases which are not described here. For internal compactions the CompactionService keeps track of the queued and running compaction tasks.
+
+![Image of Internal Compactions](https://github.com/dlmarion/accumulo-website/tree/external-compaction-design/design/internal_compactions.png)
+  
+## Configuration
+
+The CompactionManager operates on the set of online tablets hosted by the TabletServer. The CompactionManager contains a set of CompactionServices where each CompactionService is initialized with a CompactionPlanner class name, read and write rates, and other options using the following properties:
+
+  * `tserver.compaction.major.service.<service>.planner`: the fully qualified CompactionPlanner implementation class name for the service `<service>`.
+
+  * `tserver.compaction.major.service.<service>.rate.limit`: maximum number of bytes to read or write per second for all compactions in this service.
+
+  * `tserver.compaction.major.service.<service>.planner.opts.*`: other properties used to configure the planner.
+
+
+The DefaultCompactionPlanner implementation is configured to create multiple queues from the configuration where each queue is configured with file size thresholds and the number of threads. For example, the DefaultCompactionPlanner is configured by the following properties:
+
+
+  * `tserver.compaction.major.service.<service>.planner.opts.executors`: a json array where each object in the array has the fields name, maxSize, and numThreads. For example:
```

Comment:
```suggestion
  * `tserver.compaction.major.service.default.planner.opts.executors`: a json array where each object in the array has the fields name, maxSize, and numThreads. For example:
```
The default name should be "default" right?

Reply:
If you can modify the behavior of the default compaction planner by using these properties, then yes.